Data/src/test/java/org/tribuo/data/columnar/RowProcessorTest.java

```diff
@@ -170,4 +183,127 @@ public void metadataExtractorTest() {
         assertThrows(PropertyException.class, () -> new RowProcessor<>(badExtractors,weightExtractor,response,fixed,Collections.emptySet()));
     }
 
+    @Test
+    public void replaceNewlinesWithSpacesTest() {
+        final Pattern BLANK_LINES = Pattern.compile("(\n[\\s-]*\n)+");
+
+        final Function<CharSequence, CharSequence> newLiner = (CharSequence charSequence) -> {
+            if (charSequence == null || charSequence.length() == 0) {
+                return charSequence;
+            }
+            return BLANK_LINES.splitAsStream(charSequence).collect(Collectors.joining(" ⏎\n\n"));
+        };
+
+        Tokenizer tokenizer = new MungingTokenizer(new BreakIteratorTokenizer(Locale.US), newLiner);
+        TokenPipeline textPipeline = new TokenPipeline(tokenizer, 2, false);
+
+        final Map<String, FieldProcessor> fieldProcessors = new HashMap<>();
+        fieldProcessors.put("order_text", new TextFieldProcessor("order_text", textPipeline));
+
+        MockResponseProcessor response = new MockResponseProcessor("Label");
+
+        Map<String,String> row = new HashMap<>();
+        row.put("order_text", "Jimmy\n\n\n\nHoffa");
+        row.put("Label", "Sheep");
+
+        RowProcessor<MockOutput> processor = new RowProcessor<>(Collections.emptyList(),null,response,fieldProcessors,Collections.emptyMap(),Collections.emptySet(), false);
+
+        Example<MockOutput> example = processor.generateExample(row,true).get();
+
+        // Check example is extracted correctly
+        assertEquals(5,example.size());
+        assertEquals("Sheep",example.getOutput().label);
+        Iterator<Feature> featureIterator = example.iterator();
+        Feature a = featureIterator.next();
+        assertEquals("order_text@1-N=Hoffa", a.getName());
+        assertEquals(1.0, a.getValue());
+        a = featureIterator.next();
+        assertEquals("order_text@1-N=Jimmy", a.getName());
+        a = featureIterator.next();
+        assertEquals("order_text@1-N=⏎", a.getName());
+        a = featureIterator.next();
+        assertEquals("order_text@2-N=Jimmy/⏎", a.getName());
+        a = featureIterator.next();
+        assertEquals("order_text@2-N=⏎/Hoffa", a.getName());
+        assertFalse(featureIterator.hasNext());
+        a = featureIterator.next(); // weird this doesn't throw NoSuchElementException?
```

Comment:
Yep, that's a bug in the iterator on `ArrayExample`, it throws `ArrayIndexOutOfBoundsException` when exhausted rather than `NoSuchElementException`. We'll fix it.

Reply:
Fixed in #138.

Reply:
Ah, actually you were hitting a slightly different issue here, which is that the backing array for an `ArrayExample` can be larger than the size of that example, and the iterator was happy to read past size if you kept calling `next` without `hasNext` because the check wasn't correct. When #138 is integrated this line will correctly throw `NoSuchElementException`.